tasks_management/services.py

```diff
@@ -52,6 +53,40 @@ def complete_task(self, obj_data):
         except Exception as exc:
             return output_exception(model_name=self.OBJECT_TYPE.__name__, method="complete", exception=exc)
 
+    @register_service_signal('task_service.resolve_task')
+    def resolve_task(self, obj_data):
+        try:
+            self.validation_class.validate_update(self.user, **obj_data)
+            obj = self.OBJECT_TYPE.objects.get(id=obj_data['id'])
+            incoming_status = obj_data.get('business_status')
+            self._update_task_business_status(obj, incoming_status)
+            self._send_executor_action_event(obj)
+            return output_result_success({'task': model_representation(obj)})
+        except Exception as exc:
+            return output_exception(model_name=self.OBJECT_TYPE.__name__, method="resolve", exception=exc)
+
+    def _update_task_business_status(self, task, incoming_status):
+        task.business_status = {**task.business_status, **incoming_status}
+        task.save(username=self.user.login_name)
+
+    def _send_executor_action_event(self, task):
+        event_signal_string = task.executor_action_event
+        registered_signal = REGISTERED_SERVICE_SIGNALS.get(event_signal_string, None)
+        if registered_signal is not None:
+            registered_signal.register_signal(["sender", "task", "user", "business_status"])
+            registered_signal.send_signal_before(sender=self.__class__,
+                                                 task=task,
+                                                 user=self.user,
+                                                 business_status=task.business_status)
+
+    def _send_business_event(self, task):
+        data = task.data
+        event_signal_string = task.business_event
+        registered_signal = REGISTERED_SERVICE_SIGNALS.get(event_signal_string, None)
+        if registered_signal is not None:
+            registered_signal.register_signal(["sender", "obj_data"])
+            registered_signal.send_signal_before(sender=self.__class__, obj_data=data)
+
```

Comment:
This should not be required. The way tasks are set up is that all handlers for bussines and executor action event should listed to `task_service.resolve_task` AFTER signal and check for specific executor action/business events inside. That way we keep `REGISTERED_SERVICE_SIGNALS` static and don't register a signal for every possible task event to be used once and stay in the registry until the system is restarted.

Reply:
fixed